Tests fail for i386

Dear Maintainer,

during the Debian build of your package the [i386 tests failed](https://ci.debian.net/packages/g/golang-github-cloudflare-circl/testing/i386/72725648/).

I can also reproduce it locally by running `GOARCH=386 go test ./...`:

```
❯ GOARCH=386 go test ./...
go: downloading golang.org/x/crypto v0.45.0
go: downloading golang.org/x/sys v0.38.0
go: downloading github.com/bwesterb/go-ristretto v1.2.3
?   	github.com/cloudflare/circl	[no test files]
?   	github.com/cloudflare/circl/abe	[no test files]
?   	github.com/cloudflare/circl/abe/cpabe	[no test files]
ok  	github.com/cloudflare/circl/abe/cpabe/tkn20	75.185s
ok  	github.com/cloudflare/circl/abe/cpabe/tkn20/internal/dsl	0.049s
--- FAIL: TestMalformedCiphertextInternal (1.41s)
panic: runtime error: slice bounds out of range [4:3] [recovered, repanicked]

goroutine 19 [running]:
testing.tRunner.func1.2({0x821c660, 0xb89a1a0})
	/usr/lib/go-1.26/src/testing/testing.go:1974 +0x29b
testing.tRunner.func1()
	/usr/lib/go-1.26/src/testing/testing.go:1977 +0x409
panic({0x821c660, 0xb89a1a0})
	/usr/lib/go-1.26/src/runtime/panic.go:860 +0x10b
github.com/cloudflare/circl/abe/cpabe/tkn20/internal/tkn.removeLen32Prefixed({0xb998aa8, 0x91b, 0x91b})
	/home/martin/build/circl/abe/cpabe/tkn20/internal/tkn/util.go:83 +0x1e2
github.com/cloudflare/circl/abe/cpabe/tkn20/internal/tkn.CouldDecrypt({0xb998a80, 0x943, 0x943}, 0xb9a5e80)
	/home/martin/build/circl/abe/cpabe/tkn20/internal/tkn/bk.go:235 +0x103
github.com/cloudflare/circl/abe/cpabe/tkn20/internal/tkn.TestMalformedCiphertextInternal(0xb884788)
	/home/martin/build/circl/abe/cpabe/tkn20/internal/tkn/bk_test.go:363 +0x7a2
testing.tRunner(0xb884788, 0x823f13c)
	/usr/lib/go-1.26/src/testing/testing.go:2036 +0x114
created by testing.(*T).Run in goroutine 1
	/usr/lib/go-1.26/src/testing/testing.go:2101 +0x511
FAIL	github.com/cloudflare/circl/abe/cpabe/tkn20/internal/tkn	22.306s
?   	github.com/cloudflare/circl/blindsign	[no test files]
ok  	github.com/cloudflare/circl/blindsign/blindrsa	2.322s
?   	github.com/cloudflare/circl/blindsign/blindrsa/internal/common	[no test files]
?   	github.com/cloudflare/circl/blindsign/blindrsa/internal/keys	[no test files]
ok  	github.com/cloudflare/circl/blindsign/blindrsa/partiallyblindrsa	4.136s
?   	github.com/cloudflare/circl/cipher	[no test files]
ok  	github.com/cloudflare/circl/cipher/ascon	0.108s
?   	github.com/cloudflare/circl/dh	[no test files]
ok  	github.com/cloudflare/circl/dh/csidh	19.644s
ok  	github.com/cloudflare/circl/dh/curve4q	8.384s
ok  	github.com/cloudflare/circl/dh/sidh	249.899s
?   	github.com/cloudflare/circl/dh/sidh/internal/common	[no test files]
ok  	github.com/cloudflare/circl/dh/sidh/internal/p434	5.989s
ok  	github.com/cloudflare/circl/dh/sidh/internal/p503	8.018s
ok  	github.com/cloudflare/circl/dh/sidh/internal/p751	18.516s
ok  	github.com/cloudflare/circl/dh/x25519	2.737s
ok  	github.com/cloudflare/circl/dh/x448	14.082s
?   	github.com/cloudflare/circl/ecc	[no test files]
ok  	github.com/cloudflare/circl/ecc/bls12381	73.480s
ok  	github.com/cloudflare/circl/ecc/bls12381/ff	59.259s
ok  	github.com/cloudflare/circl/ecc/fourq	25.809s
ok  	github.com/cloudflare/circl/ecc/goldilocks	25.141s
ok  	github.com/cloudflare/circl/ecc/p384	6.109s
ok  	github.com/cloudflare/circl/expander	0.012s
ok  	github.com/cloudflare/circl/group	40.928s
ok  	github.com/cloudflare/circl/hpke	4.623s
?   	github.com/cloudflare/circl/internal/conv	[no test files]
?   	github.com/cloudflare/circl/internal/nist	[no test files]
ok  	github.com/cloudflare/circl/internal/sha3	0.070s
?   	github.com/cloudflare/circl/internal/test	[no test files]
?   	github.com/cloudflare/circl/kem	[no test files]
ok  	github.com/cloudflare/circl/kem/frodo	9.552s
?   	github.com/cloudflare/circl/kem/frodo/frodo640shake	[no test files]
ok  	github.com/cloudflare/circl/kem/hybrid	0.022s
ok  	github.com/cloudflare/circl/kem/kyber	0.428s
?   	github.com/cloudflare/circl/kem/kyber/kyber1024	[no test files]
?   	github.com/cloudflare/circl/kem/kyber/kyber512	[no test files]
?   	github.com/cloudflare/circl/kem/kyber/kyber768	[no test files]
ok  	github.com/cloudflare/circl/kem/mlkem	0.162s
?   	github.com/cloudflare/circl/kem/mlkem/mlkem1024	[no test files]
?   	github.com/cloudflare/circl/kem/mlkem/mlkem512	[no test files]
?   	github.com/cloudflare/circl/kem/mlkem/mlkem768	[no test files]
ok  	github.com/cloudflare/circl/kem/schemes	0.620s
?   	github.com/cloudflare/circl/kem/sike	[no test files]
?   	github.com/cloudflare/circl/kem/sike/sikep434	[no test files]
?   	github.com/cloudflare/circl/kem/sike/sikep503	[no test files]
?   	github.com/cloudflare/circl/kem/sike/sikep751	[no test files]
ok  	github.com/cloudflare/circl/kem/xwing	0.025s
ok  	github.com/cloudflare/circl/math	54.163s
ok  	github.com/cloudflare/circl/math/fp25519	0.267s
ok  	github.com/cloudflare/circl/math/fp448	0.859s
ok  	github.com/cloudflare/circl/math/mlsbset	0.180s
ok  	github.com/cloudflare/circl/math/polynomial	0.004s
ok  	github.com/cloudflare/circl/oprf	12.318s
?   	github.com/cloudflare/circl/ot	[no test files]
ok  	github.com/cloudflare/circl/ot/simot	1.666s
?   	github.com/cloudflare/circl/pke	[no test files]
?   	github.com/cloudflare/circl/pke/kyber	[no test files]
ok  	github.com/cloudflare/circl/pke/kyber/internal/common	0.619s
?   	github.com/cloudflare/circl/pke/kyber/internal/common/params	[no test files]
?   	github.com/cloudflare/circl/pke/kyber/kyber1024	[no test files]
ok  	github.com/cloudflare/circl/pke/kyber/kyber1024/internal	2.140s
?   	github.com/cloudflare/circl/pke/kyber/kyber512	[no test files]
ok  	github.com/cloudflare/circl/pke/kyber/kyber512/internal	1.196s
?   	github.com/cloudflare/circl/pke/kyber/kyber768	[no test files]
ok  	github.com/cloudflare/circl/pke/kyber/kyber768/internal	1.395s
ok  	github.com/cloudflare/circl/pki	0.060s
ok  	github.com/cloudflare/circl/secretsharing	0.128s
?   	github.com/cloudflare/circl/sign	[no test files]
ok  	github.com/cloudflare/circl/sign/bls	49.381s
ok  	github.com/cloudflare/circl/sign/dilithium	2.476s
?   	github.com/cloudflare/circl/sign/dilithium/mode2	[no test files]
ok  	github.com/cloudflare/circl/sign/dilithium/mode2/internal	2.492s
?   	github.com/cloudflare/circl/sign/dilithium/mode3	[no test files]
ok  	github.com/cloudflare/circl/sign/dilithium/mode3/internal	2.916s
?   	github.com/cloudflare/circl/sign/dilithium/mode5	[no test files]
ok  	github.com/cloudflare/circl/sign/dilithium/mode5/internal	3.143s
ok  	github.com/cloudflare/circl/sign/ed25519	5.457s
ok  	github.com/cloudflare/circl/sign/ed448	0.627s
ok  	github.com/cloudflare/circl/sign/eddilithium2	3.376s
ok  	github.com/cloudflare/circl/sign/eddilithium3	13.490s
ok  	github.com/cloudflare/circl/sign/internal/dilithium	0.247s
?   	github.com/cloudflare/circl/sign/internal/dilithium/params	[no test files]
?   	github.com/cloudflare/circl/sign/mldsa	[no test files]
ok  	github.com/cloudflare/circl/sign/mldsa/mldsa44	0.315s
ok  	github.com/cloudflare/circl/sign/mldsa/mldsa44/internal	1.862s
ok  	github.com/cloudflare/circl/sign/mldsa/mldsa65	0.387s
ok  	github.com/cloudflare/circl/sign/mldsa/mldsa65/internal	2.698s
ok  	github.com/cloudflare/circl/sign/mldsa/mldsa87	0.427s
ok  	github.com/cloudflare/circl/sign/mldsa/mldsa87/internal	3.493s
ok  	github.com/cloudflare/circl/sign/schemes	57.058s
ok  	github.com/cloudflare/circl/sign/slhdsa	108.919s
?   	github.com/cloudflare/circl/simd	[no test files]
ok  	github.com/cloudflare/circl/simd/keccakf1600	0.009s
?   	github.com/cloudflare/circl/tss	[no test files]
ok  	github.com/cloudflare/circl/tss/rsa	27.243s
?   	github.com/cloudflare/circl/tss/rsa/internal	[no test files]
?   	github.com/cloudflare/circl/tss/rsa/internal/pss	[no test files]
?   	github.com/cloudflare/circl/vdaf	[no test files]
ok  	github.com/cloudflare/circl/vdaf/prio3	0.064s
ok  	github.com/cloudflare/circl/vdaf/prio3/arith	0.252s
?   	github.com/cloudflare/circl/vdaf/prio3/arith/fp128	[no test files]
?   	github.com/cloudflare/circl/vdaf/prio3/arith/fp64	[no test files]
ok  	github.com/cloudflare/circl/vdaf/prio3/count	0.003s
ok  	github.com/cloudflare/circl/vdaf/prio3/histogram	0.008s
?   	github.com/cloudflare/circl/vdaf/prio3/internal/cursor	[no test files]
?   	github.com/cloudflare/circl/vdaf/prio3/internal/flp	[no test files]
?   	github.com/cloudflare/circl/vdaf/prio3/internal/flp_test	[no test files]
?   	github.com/cloudflare/circl/vdaf/prio3/internal/prio3	[no test files]
ok  	github.com/cloudflare/circl/vdaf/prio3/mhcv	0.008s
ok  	github.com/cloudflare/circl/vdaf/prio3/sum	0.010s
ok  	github.com/cloudflare/circl/vdaf/prio3/sumvec	0.008s
ok  	github.com/cloudflare/circl/xof	0.006s
ok  	github.com/cloudflare/circl/xof/k12	3.593s
?   	github.com/cloudflare/circl/zk	[no test files]
ok  	github.com/cloudflare/circl/zk/dl	5.104s
ok  	github.com/cloudflare/circl/zk/dleq	4.360s
ok  	github.com/cloudflare/circl/zk/qndleq	0.994s
FAIL
```

I used go 1.26:

```
go version
go version go1.26.4 linux/amd64
```

--- a/abe/cpabe/tkn20/internal/tkn/util.go
+++ b/abe/cpabe/tkn20/internal/tkn/util.go
@@ -77,6 +77,9 @@ func removeLen32Prefixed(data []byte) (next []byte, remainder []byte, err error)
 		return nil, nil, fmt.Errorf("data too short")
 	}
 	itemLen := int(binary.LittleEndian.Uint32(data))
+	if itemLen < 0 {
+		return nil, nil, fmt.Errorf("negative data length: possible overflow")
+	}
 	if (4 + itemLen) > len(data) {
 		return nil, nil, fmt.Errorf("data too short")
 	}

